management/virtualmachine/client.go

```diff
@@ -19,23 +22,42 @@ const (
 	errParamNotSpecified = "Parameter %s is not specified."
 )
 
-//NewClient is used to instantiate a new VmClient from an Azure client
+//NewClient is used to instantiate a new VirtualMachineClient from an Azure client
 func NewClient(client management.Client) VirtualMachineClient {
 	return VirtualMachineClient{client: client}
 }
 
-func (self VirtualMachineClient) CreateDeployment(role Role, cloudServiceName string) (management.OperationId, error) {
-	data, err := xml.Marshal(DeploymentRequest{
+// CreateDeployment creates a deployment and then creates a virtual machine
+// in the deployment based on the specified configuration.
+//
+// https://msdn.microsoft.com/en-us/library/azure/jj157194.aspx
+func (vm VirtualMachineClient) CreateDeployment(
+	role Role,
+	cloudServiceName string,
+	options *DeploymentOptions) (management.OperationId, error) {
```

Comment:
@svanharmelen working a lot with .NET SDK that somewhat sounds normal. :smile: As I said before, I think methods here should 1:1 match the REST API and have high-fidelity mapping and flexibility to use all features of the REST APIs. For the convenience methods we have `vmutils` package. But I'm not saying this is a religious choice or anything. That's one of the things we need to decide and unify across the codebase.